src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -581,3 +582,43 @@ def finalize(self) -> InheritedContext:
             return self
         tree = self._tree.renumber(-1)
         return InheritedContext(tree, self.found)
+
+
+T = TypeVar("T")
+
+
+class DependencyGraphWalker(abc.ABC, Generic[T]):
+
+    def __init__(self, graph: DependencyGraph, walked_paths: set[Path] | None, path_lookup: PathLookup):
+        self._graph = graph
+        self._walked_paths: set[Path] = set() if walked_paths is None else walked_paths
+        self._path_lookup = path_lookup
+
+    def walk(self) -> Iterable[T]:
```

Comment:
```suggestion
    def __iter__(self) -> Iterable[T]:
```

it's the intent, right?

Reply:
good point! done.